src/ggrc/migrations/versions/20191210_aee8f6a09419_add_workflow_state_column.py

```diff
@@ -0,0 +1,196 @@
+      c.id as c_id,
+      c.status as c_stat
+    FROM cycle_task_group_object_tasks AS c
+    JOIN relationships AS r
+    ON r.destination_type = "CycleTaskGroupObjectTask" AND r.destination_id = c.id
+      AND r.source_type = :object_type
+    JOIN cycles as cyc
+    ON cyc.id = c.cycle_id
+      WHERE cyc.is_current=1
+"""
+
+
+CTGOTS_WF_QUERY = """
+    SELECT
+      cyc.workflow_id as obj_id,
+      c.id as c_id,
+      c.status as c_stat
+    FROM cycle_task_group_object_tasks AS c
+    JOIN cycles AS cyc
+    ON cyc.id = c.cycle_id
+      WHERE cyc.is_current=1
+"""
+
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+connection = op.get_bind()
+
+
+def get_tables():
+  """Get tables in which to insert workflow_state column."""
+  tables = set()
+  for type_ in wf_models.WORKFLOW_OBJECT_TYPES:
+    model = getattr(all_models, type_)
+    tables.add(model.__tablename__)
+  tables.add(wf_models.Workflow.__tablename__)
+  return tables
+
+
+def alter_table(table_name):
+  """
+    Add 'workflow_state' column to the table.
+    Args:
+      table_name: db table name.
+  """
+  sql = """
+      ALTER TABLE `{}`
+      ADD `workflow_state` ENUM(
+          'Overdue',
+          'Verified',
+          'Finished',
+          'Assigned',
+          'In Progress'
+      )
+  """
+  connection.execute(sql.format(table_name))
+
+
+def get_object_state(states):
+  """
+    Calculate workflow_state for object.
+    Args:
+      states: set with CycleTaskGroupObjectTasks statuses
+              to which this object is mapped.
+  """
+  states = {i or ASSIGNED for i in states}
+  if OVERDUE in states:
+    return OVERDUE
+  if states in [{VERIFIED}, {FINISHED}, {ASSIGNED}]:
+    return states.pop()
+  if states == {FINISHED, VERIFIED}:
+    return FINISHED
+  return IN_PROGRESS if states else UNKNOWN_STATE
+
+
+def update_wf_state(table_name, _id, wf_state):
+  """
+    Update workflow_state value.
+    Args:
+      table_name(str): db table name.
+      _id(int): Object id.
+      wf_state(str): Calculated 'workflow_state' value.
+  """
+  sql = """
+      UPDATE `{table_name}`
+      SET `workflow_state`="{wf_state}"
+      WHERE `id`={id}
+  """
+  connection.execute(
+      sql.format(table_name=table_name, id=_id, wf_state=wf_state))
+
+
+def populate_wf_state():
+  """Populate 'workflow_state' column for objects."""
+  obj_for_revs = collections.defaultdict(list)
+  obj_types = set(wf_models.WORKFLOW_OBJECT_TYPES)
+  obj_types.add("Workflow")
+  for type_ in obj_types:
+    obj_ctgots = collections.defaultdict(set)
+
+    model = wf_models.Workflow if type_ == "Workflow" else getattr(
+        all_models, type_)
+    query = CTGOTS_WF_QUERY if type_ == "Workflow" else CTGOTS_OBJ_QUERY
+    ctgots = connection.execute(sa.text(query),
+                                object_type=type_)
+    for line in ctgots:
+      obj_ctgots[line.obj_id].add((line.c_id, line.c_stat))
+
+    for obj in obj_ctgots:
```

Comment:
`obj_ctgots`'s keys are not exactly objects but their IDs thus naming iterating variable as `obj` would not be correct. Please fix it to avoid confusion.
```suggestion
    for obj_id in obj_ctgots:
```

Reply:
Fixed.